OrbitGgp/Instance.cpp

```diff
@@ -10,59 +10,65 @@
 #include <QJsonObject>
 #include <QJsonValue>
 
+#include "OrbitGgp/Error.h"
+
 namespace OrbitGgp {
 
 namespace {
 
-Instance GetInstanceFromJson(const QJsonObject& obj) {
-  Instance inst{};
-
-  if (const QJsonValue val = obj.value("displayName"); !val.isUndefined()) {
-    inst.display_name = val.toString();
-  }
-
-  if (const QJsonValue val = obj.value("id"); !val.isUndefined()) {
-    inst.id = val.toString();
-  }
-
-  if (const QJsonValue val = obj.value("ipAddress"); !val.isUndefined()) {
-    inst.ip_address = val.toString();
+outcome::result<Instance> GetInstanceFromJson(const QJsonObject& obj) {
+  const QJsonValue display_name_val = obj.value("displayName");
+  const QJsonValue id_val = obj.value("id");
+  const QJsonValue ip_address_val = obj.value("ipAddress");
+  const QJsonValue last_updated_val = obj.value("lastUpdated");
+  const QJsonValue owner_val = obj.value("owner");
+  const QJsonValue pool_val = obj.value("pool");
+
+  if (display_name_val.isUndefined() || !display_name_val.isString() ||
```

Comment:
I think isString() implies !isUndefined()

Reply:
just checked it, it does imply it. -> done